app/src/main/kotlin/com/wire/android/ui/home/conversations/ConversationViewModel.kt

```diff
@@ -469,18 +473,44 @@ class ConversationViewModel @Inject constructor(
         }
     }
 
-    private suspend fun fromMessageModelToMessageContent(message: Message): MessageContent? =
+    private fun MemberDetails.toSystemMessageName(lowerCaseSelfLabel: Boolean = false) = when (this) {
+        is MemberDetails.Other -> this.name?.let { UIText.DynamicString(it) } ?: UIText.StringResource(R.string.member_name_deleted_label)
+        is MemberDetails.Self -> UIText.StringResource(
+            if (lowerCaseSelfLabel) R.string.member_name_you_label_lowercase
+            else R.string.member_name_you_label_titlecase
+        )
+    }
+
```

Comment:
Can you move this to a mapper class or somewhere else ? and cover it with unit test
 It's  not the viewModel responsability to make mapping

Reply:
Sure, I created separate mapper classes, I noticed that Mateusz is also moving them so I did it similarly to make resolving conflicts easier 😄 I also provided tests for all these mapper functions 💪